holidays/countries/norway.py

```diff
@@ -37,61 +36,74 @@ class Norway(HolidayBase):
-                rrule.WEEKLY, byweekday=SU, dtstart=year_first_day
-            ).between(year_first_day, year_last_day, inc=True)
-            for sunday in sundays:
-                self[sunday.date()] = "Søndag"
+        # Good Friday.
+        self._add_good_friday(tr("Langfredag"))
 
-        # ========= Static holidays =========
-        self[date(year, JAN, 1)] = "Første nyttårsdag"
+        # Easter Sunday.
+        self._add_easter_sunday(tr("Første påskedag"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Andre påskedag"))
 
         # Source: https://lovdata.no/dokument/NL/lov/1947-04-26-1
         if year >= 1947:
-            self[date(year, MAY, 1)] = "Arbeidernes dag"
-            self[date(year, MAY, 17)] = "Grunnlovsdag"
+            # Labour Day.
+            self._add_labour_day(tr("Arbeidernes dag"))
+
+            # Constitution Day.
+            self._add_holiday(tr("Grunnlovsdag"), MAY, 17)
+
+        # Ascension Day.
+        self._add_ascension_thursday(tr("Kristi himmelfartsdag"))
+
+        # Whit Sunday.
+        self._add_whit_sunday(tr("Første pinsedag"))
+
+        # Whit Monday.
+        self._add_whit_monday(tr("Andre pinsedag"))
 
         # According to https://no.wikipedia.org/wiki/F%C3%B8rste_juledag,
         # these dates are only valid from year > 1700
         # Wikipedia has no source for the statement, so leaving this be for now
-        self[date(year, DEC, 25)] = "Første juledag"
-        self[date(year, DEC, 26)] = "Andre juledag"
-
-        # ========= Moving holidays =========
-        # NOTE: These are probably subject to the same > 1700
-        # restriction as the above dates. The only source I could find for how
-        # long Easter has been celebrated in Norway was
-        # https://www.hf.uio.no/ikos/tjenester/kunnskap/samlinger/norsk-folkeminnesamling/livs-og-arshoytider/paske.html
-        # which says
-        # "(...) has been celebrated for over 1000 years (...)" (in Norway)
-
-        easter_date = easter(year)
-        self[easter_date + td(days=-3)] = "Skjærtorsdag"
-        self[easter_date + td(days=-2)] = "Langfredag"
-        self[easter_date] = "Første påskedag"
-        self[easter_date + td(days=+1)] = "Andre påskedag"
-        self[easter_date + td(days=+39)] = "Kristi himmelfartsdag"
-        self[easter_date + td(days=+49)] = "Første pinsedag"
-        self[easter_date + td(days=+50)] = "Andre pinsedag"
+
+        # Christmas Day.
+        self._add_christmas_day(tr("Første juledag"))
+
+        # Second Christmas Day.
+        self._add_christmas_day_two(tr("Andre juledag"))
+
+        if self.include_sundays:
+            # Optionally add all Sundays of the year.
+            begin = _get_nth_weekday_of_month(1, SUN, JAN, year)
+            end = date(year, DEC, 31)
+            for dt in (
+                begin + td(days=n) for n in range(0, (end - begin).days + 1, 7)
+            ):
+                if not self.get(dt):
```

Comment:
In fact, I don't quite understand how Sundays in Sweden and Norway differ from Sundays in other countries. They are non working days according to labor laws almost everywhere (except for some Arab countries).

In general, a parameter equal to `include_sundays` would be useful for other countries as well. It would help to answer the question "is it a working day?"